From c19e513c48c9c03ff29e51ba5e35d866cea449ec Mon Sep 17 00:00:00 2001
From: Cristy <urban-warrior@imagemagick.org>
Date: Fri, 3 Jun 2016 20:16:40 -0400
Subject: [PATCH] Check return of write function when writting tiff file

Do not checking return of write function could lead to incomplete files

bug: https://github.com/ImageMagick/ImageMagick/issues/196
bug-debian: https://bugs.debian.org/845196
---
 coders/tiff.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

Index: imagemagick-6.7.7.10/coders/tiff.c
===================================================================
--- imagemagick-6.7.7.10.orig/coders/tiff.c	2016-11-29 08:44:22.257505426 -0500
+++ imagemagick-6.7.7.10/coders/tiff.c	2016-11-29 08:44:22.253505382 -0500
@@ -370,8 +370,10 @@
   length=fwrite("\000\000\000\000",1,4,file);
   length=WriteLSBLong(file,(size_t) (image->x_resolution+0.5));
   length=WriteLSBLong(file,1);
+  status=MagickTrue;
   for (length=0; (c=ReadBlobByte(image)) != EOF; length++)
-    (void) fputc(c,file);
+    if (fputc(c,file) != c)
+      status=MagickFalse;
   offset=(ssize_t) fseek(file,(ssize_t) offset,SEEK_SET);
   length=WriteLSBLong(file,(unsigned int) length);
   (void) fclose(file);
@@ -393,6 +395,8 @@
       (void) CopyMagickString(image->magick,"GROUP4",MaxTextExtent);
     }
   (void) RelinquishUniqueFileResource(filename);
+  if (status == MagickFalse)
+    image=DestroyImage(image);
   return(image);
 }
 #endif
